test/functional/test_framework/wallet.py

```diff
@@ -67,13 +67,26 @@ def send_self_transfer(self, *, fee_rate=Decimal("0.003"), from_node, utxo_to_sp
         tx = CTransaction()
         tx.vin = [CTxIn(COutPoint(int(utxo_to_spend['txid'], 16), utxo_to_spend['vout']))]
         tx.vout = [CTxOut(int(send_value * COIN), self._scriptPubKey)]
+        tx.nLockTime = locktime
         tx.wit.vtxinwit = [CTxInWitness()]
         tx.wit.vtxinwit[0].scriptWitness.stack = [CScript([OP_TRUE])]
         tx_hex = tx.serialize().hex()
 
         tx_info = from_node.testmempoolaccept([tx_hex])[0]
-        self._utxos.append({'txid': tx_info['txid'], 'vout': 0, 'value': send_value})
+        if tx_info['allowed']:
+            assert_equal(tx_info['vsize'], vsize)
+            assert_equal(tx_info['fees']['base'], fee)
+
+        new_utxo = {'txid': tx_info['txid'], 'vout': 0, 'value': send_value}
+        return tx_hex,  new_utxo
+
+    def send_self_transfer(self, *, fee_rate=Decimal("0.003"), from_node, utxo_to_spend=None, locktime=0):
+        """Create and send a tx with the specified fee_rate. Fee may be exact or at most one satoshi higher than needed."""
+
+        tx_hex, new_utxo = self.create_self_transfer(fee_rate=fee_rate, from_node=from_node, utxo_to_spend=utxo_to_spend, locktime=locktime)
+
+        tx_info = from_node.testmempoolaccept([tx_hex])[0]
```

Comment:
this is not needed anymore i believe

Reply:
I believe `tx_info` is being reused used in line 92

Reply:
Also is there a need to recheck `testmempoolaccept`? Every time we call `send_self_transfer` it checks the mempool acceptance through `create_self_transfer` so, can we just return proper values and remove the additional check?

Reply:
good catch! I replaced it with `testmempoolaccept` with `FromHex` to get the txid and wtxid.